gsoc-2024-project-list.md

```diff
@@ -0,0 +1,218 @@
+
+**Project size**: medium
+
+**Mentors**: Roman Ivanov, Nick Mancuso, Vyom Yadav
+
+**Description**:
+
+This project is designed to tackle the persistent challenge of
+maintaining accurate and current documentation in our dynamic development environment. Acknowledging the limitations of
+manual documentation processes, this initiative introduces automation to streamline content creation, with a focus on
+ensuring consistent formats and robust verification checks. The project's goal is to provide users with reliable,
+standardized, and regularly updated information while equipping contributors with templates and automated tools to
+simplify the incorporation of details for new modules. By elevating documentation practices, this project aligns with
+industry best practices, fostering clarity for both users and contributors within the Checkstyle project.
+
+**Deliverables**:
+
+- Resolution of edge cases in documentation generation
+- Extend and make consistent all check usage examples
+- Reduce/eliminate manual documentation updates for examples
+- Introduce checks to ensure that all configuration options are covered in examples
+
+***
+
+### Project Name: Refine Google Style Guide Implementation
+
+**Skills required**: Java, basic understanding of testing principles, basic understanding of static analysis
+
+**Project goal**: improve quality of google style guide implementation
+
+**Project size**: large
+
+**Mentors**: Roman Ivanov, Richard Veach
+
+**Description**:
+
+Checkstyle boasts a robust implementation of
+the [Google style](https://github.com/google/styleguide/commits/gh-pages/javaguide.html), well-documented on
+our [coverage page](https://checkstyle.org/google_style.html) specifying the supported style version. To maintain
+alignment with updates in the Google style guide, we aim to systematically review changes such
+as [this commit](https://github.com/google/styleguide/commit/49488412b2f59843fce2433bd834a3fd9700604e) and adapt our
+Checks configuration accordingly. Addressing reported defects and user issues regarding style guide mismatches is
+pivotal. All concerns reported for Modules/Checks present in
+the [Google Style config](https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml)
+will be thoroughly reviewed, appropriately labeled for easy filtration, and promptly rectified. Conceptually, a
+refactoring of our integration tests is imperative; transitioning from a per-module approach to a chapter-wise testing
+methodology aligned with the guideline structure. By mapping chapter requirements to sets of modules loaded from
+google_checks.xml, we can conduct comprehensive validations over input files, presenting a consolidated result.
+
+**Deliverables**:
+
+- Review and update Google Style config to most recent content of style guide
+- Resolve known issues with modules
+- Refactor integration tests to be chapter-wise
+- Resolve all reported issues with Google Style config
```

Comment:
moved to https://github.com/checkstyle/checkstyle/wiki/Checkstyle-GSoC-2024-Project-Ideas#project-name-refine-google-style-guide-implementation